### DISCUSSION TITLE
Vuetify child component form validation example

### DISCUSSION THREAD

**Discussion by james-boswell** [vuetifyjs/vuetify#11934] [General]
Hi

I was wondering if someone could provide a working example of using Vuetify form validation where the parent component contains the v-form with a submit button and it also contains a custom component which, for example, wraps a v-textarea which is required.

Similar to the "Validation with submit & clear" example here:
https://vuetifyjs.com/en/components/forms/

but using a custom child component.

Thanks

**Comment by JRB-y** [comment]
No answer for this?

**Comment by tsung-jen1503** [comment]
I am facing the same situation. I have a parent component with v-form and v-btn, and multiple custom child components contain several v-select and v-text-field to be validated (only one layer). The built-in validation for v-form is not worked with a parent and child relation. Could someone provide a solution? Thanks a lot!

**Comment by angeles-ricardo-89** [comment]
I am in the same situation, since there is not response or an explicit answer about this, it looks like it should be quite obvious for everyone else but me. I think I have to keep searching, I'll post here whatever I find.

**Reply by KaelWD** [reply]
Nobody here has said what they've actually tried, it should just work.

**Reply by angeles-ricardo-89** [reply]
Hi, I've just been able to do this by extending Validatable from vuetify2 mixins. I tried to extend VInput first, but all my attempts failed.

Basically I followed these steps:
1.  Extend Validatable.
2.  emit 'input' signal for every model change using watchers.
3.  provide rules array to rules property of the custom component.
4.  Add text to the custom component template in order to show the errors in errorBucket produced by the rules.

It's simple to implement, but not so easy to come up with the idea, I'd recommend people to examine Validatable source code in order to understand how it works. I'm not sure if I've done this in the right way, if there is a better way to do it I'd apreciate any advice, thanks.

**Reply by KaelWD** [reply]
If you're using v2: https://github.com/vuetifyjs/vuetify/issues/3464#issuecomment-370240024

**Comment by Naedri** [comment]
Hello,

I tried to provide an example in VueJS v3.5 and Vuetify v3.7 : [Playground](https://play.vuetifyjs.com/#eNqlVl2O2zYQvsrEfZAN2NamRVvAkNU6SQP0YdEgDfbFNrC0RFuEZVKlKBuu4zP0uTcp0Nv0Ar1CZ0jqb9dIkfZhLXE4P9/MfDPa5WVQ6iRcFMX0WPHBbBC9mExW8kMmSkj5QQErCmDiUIJRUOTsDCdhMjAZh8eHH2VRmUdI1KFQkkszhpUUEpROuSb9vVQnyPAP340Wux2KhSnhyHKRMiOUBF3lvIRTxqX1ycpSJYIZnsLjcXJQKc/Rf8bkjpfTlZxM4pVcycjwA2IxHE8A0XGCIO2rPSRKGiYk116Ewuyr+P4MmyrZcwNqC0e+44ZtMHQU4l2j945pTKPGx3+S89VAUJKrAYR1hPBpCJI4BFHYQYbHMtGiMFByUzmAAiulDfg4W60OEExDd6QGBOTDWaH+YGyb015Tf/q5F3Ek4sulRQzfdeDDDIJKpnyLWNPgeo1CEQNVNQodHiyXy8+mpvm2TdiJfBNQ7KpXy2d1kImiIrUhGwXbWbyyTyetK5iKY1vz15nI04d7ilJHm7lYnaAwKzNVoIAevWbUvqgJFnmd18ZIZGauNFqdmEZi7tDw+yQXyZ5gcezKQ0PE1SB+T5ION8kjOvmMtl6ognD1fXXtbC67edaN78h898kA2VUaKLQqSpjjFFL33tFpePFNaao9Ay8CMOeCz+BnnDO5G9dCNGZVbmbQsMBfXe3zOupGdLM4h6WPAvMYfwReMZlwHJt7VsDHjxBIZWCnVAoIggXeo1OfluJXDjF82Shyqapddksrgq+tllEKDlXiddZdSH5k51TYofMg+YmADD1KgGWAGCBg2ogkU3serJv0l8GGrjZaJUgF0dysMW+AXvJELMq946drie8J01oZekuZ2CtJb1wi/6zmlkvJ86CH3vLRg5dVntuovtZ9+qESK88ygeGICuSbKrYwtD6m2PGKj4CdmPBunWj6xM/Qhrje2iJPuPZ8lfRH5otvX73+5u1bGr1UlLQrU5S+SJhcpGl3kBgdYxS28/Lc2eKHN29uOnuP35gj7w+mk+A80kvPa0UrQtCSqhdDvQdygTdbG0xgWrTk292x52cvX96t0eZywc8QHl6ugXZiLvwGqfLPGnb67lX0qepP/A32utmztR8GThqMIAyp13/9/ufff/zWNbs4Ml5vDz/dPRv7hdbs3NBe818qoXk6w69uxT8x8XYM500iQ8c+B9ATjFR6NlS5Dwr7/dyQoC3JwNm6KetauZb+qyFM4CUZt+aOdjcMW5vIepnmXO5M1onckOyT1jHc+YDbSiZ2JJHYqOYL7Xzh7nlV97TeQ91q2bB2RTm1KbJm2BTMVxT/ndELM7wbjdta9sy7Hl0gH7RPGDvnPcRudP43aGQpN9zjdrV7Cv0/IO1upOuY1qH9Wf8DgbyDgw==), in which a `Parent` component hosts two components :

- a `VInput` component : 
    - responsible to display errors
- a `ChildVModel` component : 
    - responsible to display a list of items initialized by the `Parent` 
    - responsible to update a list of items using v-model

I hope it could help :)
